include/utilities/FileStreamHandler.hpp

```diff
@@ -1,16 +1,20 @@
 #ifndef NGEN_FILE_STREAM_HANDLER_HPP
 #define NGEN_FILE_STREAM_HANDLER_HPP
 
-#include "StreamHandler.hpp""
+#include "StreamHandler.hpp"
 
 namespace utils
 {
-    class FileStreamHandler() : public StreamHandler
+    class FileStreamHandler : public StreamHandler
     {
-            FileStreamHandler(const char*) : StreamHandler()
+      public:
+            FileStreamHandler(const char* path) : StreamHandler()
             {
-                output_stream = make_shared(new(std::fstream(path)));
```

Comment:
Is it possible that this change is why you where having issues with fstream objects>

Reply:
This wouldn't compile.